proxy/src/main/java/com/wavefront/agent/ChannelStringHandler.java

```diff
@@ -69,6 +88,17 @@ protected void channelRead0(ChannelHandlerContext ctx, String msg) throws Except
       }
     }
     // if msg does not match metadata keywords then treat it as a point
+    // use data rate to determine sampling rate
+    // logging includes the source host and port
+    if(logRawDataRate >= 1.0d || (logRawDataRate > 0.0d && RANDOM.nextDouble() < logRawDataRate)) {
+      if(ctx.channel().remoteAddress() != null) {
+        String host = ((InetSocketAddress) ctx.channel().remoteAddress()).getHostName();
```

Comment:
getHostName() is a reverse DNS lookup, which is a very expensive operation. In high traffic environments one getHostName() _per incoming connection_ has caused issues previously, so one rDNS lookup per logged point is virtually guaranteed to cause performance issues (and unleash the wrath of your network engineer as an added bonus).
Just logging remote IP (ctx.channel().remoteAddress().getAddress().getHostAddress()) should be fine, or cache rDNS lookup results like in PushAgent.reverseDnsCache

Reply:
Ok, thanks for the comment. makes sense.

Reply:
changed the delimiter ':' to '>', as the local port could be confused as remote port. Removed remote port in the log.